packages/batcher/webserver/src/index.ts

```diff
@@ -405,49 +406,66 @@ async function initializeServer(
           return;
         }
 
-        if (validity === 0) {
+        if (validity !== 0) {
+          res.status(400).json({
+            success: false,
+            message: errorCodeToMessage(validity),
+            code: validity,
+          });
+          return;
+        }
+        const payload = {
+          address_type: addressType,
+          user_address: userAddress,
+          game_input: gameInput,
+          millisecond_timestamp: millisecondTimestamp,
+          user_signature: userSignature,
+        };
+
+        let weiUsed = BigInt(0);
+        if (ENV.BATCHER_PAYMENT_ENABLED) {
           try {
-            await reCaptchaValidation(captcha);
-            unsetWebserverClosed();
-            await insertStateValidating.run({ input_hash: hash }, pool);
-            await insertUnvalidatedInput.run(
-              {
-                address_type: addressType,
-                user_address: userAddress,
-                game_input: gameInput,
-                millisecond_timestamp: millisecondTimestamp,
-                user_signature: userSignature,
-              },
-              pool
-            );
-            setWebserverClosed();
+            weiUsed = await BatcherPayment.estimateGasFeeInWei(JSON.stringify(payload));
```

Comment:
I think this code path is triggered even in non-EVM cases

Reply:
Yes, it will fail as it has not been implemented.
Not sure what to do in this case.

https://github.com/PaimaStudios/paima-engine/pull/422/files/1ff0cf5cb1c7a7d0960632c8b353acda31e079c4#diff-a879925905151f317071cbb2b6633f93de28d49c8937bcd881c4b4efd933865bR64

Reply:
Removed reference to `wei` units